app/modules/intelligence/agents/custom_agents/custom_agents_service.py

```
@@ -676,8 +728,8 @@ async def create_agent_from_prompt(
         try:
             available_tools = ToolService(self.db, user_id).list_tools()
             tool_ids = [tool.id for tool in available_tools]
-        except Exception as e:
-            logger.error(f"Error fetching available tools: {str(e)}")
+        except Exception:
+            logger.exception("Error fetching available tools", user_id=user_id)
             raise HTTPException(
                 status_code=500, detail="Failed to fetch available tools"
             )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)